actions/global_actions.tsx

```diff
@@ -191,22 +191,24 @@ export function showMobileSubMenuModal(elements: any[]) { // TODO Use more speci
     dispatch(openModal(submenuModalData));
 }
 
-export function sendEphemeralPost(message: string, channelId?: string, parentId?: string, userId?: string): void {
-    const timestamp = Utils.getTimestamp();
-    const post = {
-        id: Utils.generateId(),
-        user_id: userId || '0',
-        channel_id: channelId || getCurrentChannelId(getState()),
-        message,
-        type: PostTypes.EPHEMERAL,
-        create_at: timestamp,
-        update_at: timestamp,
-        root_id: parentId,
-        parent_id: parentId,
-        props: {},
-    };
-
-    dispatch(handleNewPost(post));
+export function sendEphemeralPost(message: string, channelId?: string, parentId?: string, userId?: string): ActionFunc {
```

Comment:
If this is not accessing the Global Store, I guess it is no longer a Global Action. Does it make sense to be in the Global Actions file?

Reply:
@larkox Moving the action to another file caused issues with mocking in `actions/command.test.js`. I was able to resolve the mocking issues, but it complicated tests unrelated to the changes in this PR, so I'm keeping the action in `global_actions.ts`.

FWIW there is one other function that is "dispatchable" in that file 🙂  If any function in the future wants to import and use `sendEphemeralPost`, it will "fail fast" so 1/5 I think the lack of distinction is okay.

Reply:
I'm fine leaving it here for now. Ideally, these would be better compartmentalized, but we don't have a lot of good solutions like to the mocking problem. We're planning on investigating [sagas](https://redux-saga.js.org/) as an alternative to thunk actions (the ones that return functions) in part because they solve that exact problem.